opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -503,9 +500,107 @@ message DoubleHistogramDataPoint {
+  // It will only need to record the exponential bound parameters.
+  message ExponentialBounds {
+    double reference = 1;
+    double base = 2;
+
+    // The bound sequence starts with negative values generated via:
+    // for (int i = num_of_bounds_for_negative_numbers - 1; i >= 0; i--)
+    //    int exponent = i + index_offset_for_negative_numbers;
+    //    boundList.add( -1 * reference * power(base, exponent));
+    //
+    // Example of reference=1, base=2, index_offset_for_negative_numbers=-2, num_of_bounds_for_negative_numbers=6
+    // bounds:    -8 -4 -2 -1 -.5 -.25
+    // exponent:   3  2  1  0 -1  -2
+    // Note that the loop on i goes down, so that negative numbers with larger absolute values
+    // (but smaller arithmetic values) are added first.
+
+    sint32 index_offset_for_negative_numbers = 3;
+    uint32 num_of_bounds_for_negative_numbers = 4;
+
+    // If has_counter_for_zero is true, add a bound of zero using:
+    //    boundList.add(0);
+    bool has_counter_for_zero = 5;
+
+    // Positive bounds follow the zero bound. Positive bounds are generated via:
+    // for (int i = 0; i < num_of_bounds_for_positive_numbers; i++)
+    //    int exponent = i + index_offset_for_positive_numbers;
+    //    boundList.add( reference * power(base, exponent));
+    //
+    // Example of reference=1, base=2, index_offset_for_positive_numbers=-3, num_of_bounds_for_positive_numbers=8
+    // bound:    .125 .25 .5 1 2 4 8 16
+    // exponent: -3   -2  -1 0 1 2 3 4
+
+    sint32 index_offset_for_positive_numbers = 6;
+    uint32 num_of_bounds_for_positive_numbers = 7;
+  }
+
+  // In LogLinearBounds, each exponential bucket is divided linearly into multiple subbuckets.
+  // LogLinearBounds is an extension of ExponentialBounds. Similar to ExponentialBounds,
+  // the bound sequence is stitched together using negative bounds, zero bound, and positive bounds.
+  //
+  // Example of LogLinearBounds with num_of_linear_subbuckets=4 on ExponentialBounds of
+  //    reference=1, base=2, index_offset_for_positive_numbers=-4, num_of_bounds_for_positive_numbers=15,
+  //
+  // bound:  .5 .625 .75 .875 1 1.25 1.5 1.75 2 2.5 3 3.5 4 5 6
+  // index:  -4 -3   -2  -1   0 1    2   3    4 5   6 7   8 9 10
+  //
+  // Note: Index need not start or end on multiples of num_of_linear_subbuckets.
+  //
+  // Positive bounds are computed as:
+  // for (int count = 0; count < num_of_bounds_for_positive_numbers; count++)
+  //    int i = count + index_offset_for_positive_numbers;
+  //    int exponent;
+  //    int subbucketNumber; // In the range of [0, num_of_linear_subbuckets - 1]
+  //    if (i >= 0) {
+  //        exponent = i / num_of_linear_subbuckets;
+  //        subbucketNumber = i % num_of_linear_subbuckets;
+  //    } else {
+  //        exponent = (i - num_of_linear_subbuckets + 1) / num_of_linear_subbuckets; // Round down toward -infinity
+  //        subbucketNumber = i - exponent * num_of_linear_subbuckets;
+  //    }
```

Comment:
On line 586: A comment like 
```
// exponent is negative, same as (i + abs(exponent) * num_of_linear_subbuckets)
```

I'm not sure that helps, but it may have helped me.